python/lib/modeldata/dmod/modeldata/data/object_store_dataset.py

```diff
@@ -156,32 +157,17 @@ def __init__(self, obj_store_host_str: str, access_key: Optional[str] = None, se
         # TODO: add checks to ensure all datasets passed to this type are ObjectStoreDataset
         self._obj_store_host_str = obj_store_host_str
         # TODO (later): may need to look at turning this back on
-        self._client = Minio(obj_store_host_str, access_key=access_key, secret_key=secret_key, secure=False)
-        # For any buckets that have the standard serialized object (i.e., were for datasets previously), reload them
-        for bucket_name in self.list_buckets():
-            self._load_from_existing_bucket(bucket_name)
-
-    def _decode_object_name_to_file_path(self, object_name: str) -> str:
-        """
-        Reverse the object name encoding of nested file names.
-
-        This essentially just decodes the encoding performed by ::method:`_push_file`.
-
-        Parameters
-        ----------
-        object_name : str
-            The name of the object storing some previously uploaded file data.
-
-        Returns
-        -------
-        str
-            The decoded file name that reflects any subdirectory structure.
-
-        See Also
-        -------
-        ::method:`_push_file`
-        """
-        return "/".join(object_name.split(self._OBJECT_NAME_SEPARATOR))
+        try:
+            self._client = Minio(endpoint=obj_store_host_str, access_key=access_key, secret_key=secret_key, secure=False)
```

Comment:
Another non-showstopper that jumped out at me. Yeah, it'll make things more complicated, but I wonder if an external function or dependency injection might help decouple a hard reliance on Minio here. I'm not sure if there are super specific Minio functionality used throughout, so I'm not really qualified to say, but if Minio fulfills the S3 protocol/API/whatever, couldn't other implementations work here? That would allow everyone to use different object stores based on environment. Changing this up later down the line would allow this to be used with Amazon S3, google cloud, CEPH, etc.

Not a showstopper since **IF** it is reasonable to address, it should be a problem for future DMOD considering the work required. There are a ton of other priorities right now.